x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java

```diff
@@ -194,7 +194,7 @@ public void writeTo(StreamOutput out) throws IOException {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         {
-            if (Strings.hasText(asyncExecutionId)) {
```

Comment:
I wonder if using different class hierarchy instead of dedicated properties might help encapsulate the difference in response between sync/async better.
The `hasId` is obscure - is it used anywhere else?

Reply:
> I wonder if using different class hierarchy instead of dedicated properties might help encapsulate the difference in response between sync/async better.

That'd be ideal, but I didn't find a way to cross the hierarchies.

> The hasId is obscure - is it used anywhere else?

`hasId()` is also used in SqlResponseFormatter and TextFormat, yes. It is a bit obscure and I hesitated using it, but eventually did since there is a `hasCursor()` (which returns the `cursor` field) and there is already the `id()` method. Now the `id()` method is maybe also generic, but I wanted to keep the symmetry with `EqlSearchResponse`.
(We could change them in both classes to something like `asyncId()` & co., but I'd do that subsequently, if needed.)